include/h2o.h

```diff
@@ -2356,6 +2380,16 @@ void h2o_headers_append_command(h2o_headers_command_t **cmds, int cmd, h2o_heade
  */
 void h2o_rewrite_headers(h2o_mem_pool_t *pool, h2o_headers_t *headers, h2o_headers_command_t *cmd);
 
+
+/**
+ * extract header name from given string
+ */
+int h2o_headers_extract_name(const char *src, size_t len, h2o_iovec_t **name);
+/**
+ * extract header name and value from given string
+ */
+int h2o_headers_extract_name_value(const char *src, size_t len, h2o_iovec_t **name, h2o_iovec_t *value);
```

Comment:
I think it makes sense to reuse these functions.

But the names are confusing, as they are not dealing with headers (list of headers). How about moving them to string_.h / string.c, and calling them `h2o_extract_header_name`, `h2o_extract_header_name_and_value`.

Also, I would appreciate writing down what the return values mean and what name and values represent (are they allocated? are they tokens?).

Reply:
Addressed in https://github.com/h2o/h2o/pull/3405/commits/5bb1798f9539722d43f30e582bc48df86a46d31a